src/librustc/ty/maps/on_disk_cache.rs

```diff
@@ -49,6 +49,7 @@ pub struct OnDiskCache<'sess> {
 
     prev_cnums: Vec<(u32, String, CrateDisambiguator)>,
     cnum_map: RefCell<Option<IndexVec<CrateNum, Option<CrateNum>>>>,
+    prev_def_path_tables: Vec<DefPathTable>,
```

Comment:
Does this signal a shift away from the "reverse lookup by hash" strategy?

Reply:
No, the `DefPathTables` are stored for facilitating the "reverse lookup by hash". The implementation in this PR does the following:
- Store a `DefId` unmodified in the cache.
- When loading, we need to 
  1. map the serialized `DefId` to the session-independent `DefPathHash` and then
  2. look up the new `DefId` by this `DefPathHash`.

In order to be able to do step (i), we need the *old* `DefPathTable` because it maps `DefIndices` to `DefPathHashes`. 

But,... I'm not surprised that this caught your eye. There is an alternative approach that is simpler to implement and maybe more robust: Map `DefIds` and `DefIndices` to `DefPathHashes` *at serialization time* and thus get rid of the requirement to store the old `DefPathTables`. I'm not sure if this affects performance and space requirements in a good or in a bad way. But I'd like to implement it soonish and measure the impact.

Reply:
OK, I see.